shared/src/main/scala/mlscript/TypeSimplifier.scala

```diff
@@ -268,7 +268,7 @@ trait TypeSimplifier { self: Typer =>
                 })(noProv)
                 println(s"typeRef ${typeRef}")
                 
-                val clsFields = fieldsOf(typeRef.expandWith(paramTags = true, selfTy = false), paramTags = true)
+                val clsFields = fieldsOf(typeRef.expandWith(paramTags = true, selfTy = false, pol.getOrElse(true)), paramTags = true)
```

Comment:
I don't know how to fix it yet but this is definitely wrong.
```suggestion
                val clsFields = fieldsOf(typeRef.expandWith(paramTags = true, selfTy = false, pol.getOrElse(die/*TODO*/)), paramTags = true)
```